src/routes/api/rpc/[provider]/[network]/+server.ts

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { RequestHandler } from './$types';
+import assert from '$lib/utils/assert';
+import getOptionalEnvVar from '$lib/utils/get-optional-env-var/public';
+
+const providers = ['infura', 'alchemy', 'glif'] as const;
+const networkNames = [
+  'mainnet',
+  'sepolia',
+  'polygon-amoy',
+  'optimism-sepolia',
+  'base-sepolia',
+  'filecoin-mainnet',
+] as const;
+
+const alchemyNetworkMap: Record<
+  Exclude<(typeof networkNames)[number], 'filecoin-mainnet'>,
+  string
+> = {
+  mainnet: 'eth-mainnet',
+  sepolia: 'eth-sepolia',
+  'polygon-amoy': 'polygon-amoy',
+  'optimism-sepolia': 'opt-sepolia',
+  'base-sepolia': 'base-sepolia',
+};
+
+const isValidProvider = (provider: any): provider is (typeof providers)[number] =>
+  providers.includes(provider);
+const isValidNetwork = (network: any): network is (typeof networkNames)[number] =>
+  networkNames.includes(network);
+
+/** Proxies all requests to the RPC provider and injects API access token */
+export const fallback: RequestHandler = async ({ request, params, fetch }) => {
+  const body = await request.text();
+
+  const { provider, network } = params;
+  assert(isValidProvider(provider), 'Invalid provider name');
+  assert(isValidNetwork(network), 'Invalid network name');
+
+  let rpcUrl: string;
+  const headers = new Headers();
+
+  if (provider === 'infura') {
+    const infuraKey = getOptionalEnvVar('INFURA_KEY');
+    assert(infuraKey, 'INFURA_KEY is required');
+    rpcUrl = `https://${network}.infura.io/v3/${infuraKey}`;
+  } else if (provider === 'alchemy') {
+    const alchemyKey = getOptionalEnvVar('ALCHEMY_KEY');
+    assert(alchemyKey, 'ALCHEMY_KEY is required');
+    rpcUrl = `https://${alchemyNetworkMap[network as keyof typeof alchemyNetworkMap]}.g.alchemy.com/v2/${alchemyKey}`;
```

Comment:
The typecast here could be unsafe if `provider` is `alchemy` but `network` is `filecoin-mainnet`. Generally I think typecasting should be avoided unless absolutely necessary. I would assert explicitly here that `network` is included in `alchemyNetworkMap`

Reply:
oh also, i think the entire block would be neater as a `switch` statement